backend/src/git.rs

```diff
@@ -298,7 +309,40 @@ impl Interface {
     // }
 
     /// A code level re-implementation of `git commit`.
-    ///
+    /// A function used to checkout or create a new branch based on the name.
+    pub fn checkout_or_create_branch(repo: &Repository, branch_name: &str) -> Result<()> {
+        let head = repo.head()?;
+        let commit = head.peel_to_commit()?;
+    
+        match repo.find_branch(branch_name, BranchType::Local) {
+            Ok(_) => {
+                // If the branch exists, check it out
+                repo.set_head(&format!("refs/heads/{}", branch_name)).map_err(|e| {
+                    error!("Failed to set head to branch {branch_name}: {e:?}");
+                    e
+                })?;
+            }
+            Err(_e) => {
+                // If the branch does not exist, create it
+                match repo.branch(branch_name, &commit, false) {
+                    Ok(_) => {
+                        // Now check out the newly created branch
+                        repo.set_head(&format!("refs/heads/{}", branch_name)).map_err(|e| {
+                            error!("Failed to set head to new branch {branch_name}: {e:?}");
+                            e
+                        })?;
+                    }
+                    Err(create_err) => {
```

Comment:
Same here, `.wrap_err`. 

If you wanted, this whole match block could probably be refactored into a linear control flow construct based around `?`, rather than the equivalent of catching an error than immediately throwing it again.

If you'd like logs to be emitted when a function returns an error, you can actually annotate the function with
`#[tracing::instrument(err)]`

https://docs.rs/tracing/latest/tracing/attr.instrument.html#:~:text=If%20the%20function%20returns%20a%20Result%3CT%2C%20E%3E%20and%20E%20implements%20std%3A%3Afmt%3A%3ADisplay%2C%20adding%20err%20or%20err(Display)%20will%20emit%20error%20events%20when%20the%20function%20returns